In [32]:
# Welcome to your new notebook
# Type here in the cell editor to add code!
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *

StatementMeta(, fab74371-09bd-4adf-91a7-69aab161d09f, 34, Finished, Available, Finished)

In [33]:
%%sql
create table if not EXISTS Gold_Date
(
BusinessDate Date,
Business_Year int,
Business_Month int,
Business_Quarter int,
Business_Week int
)
using DELTA
PARTITIONED by (Business_Year)

StatementMeta(, fab74371-09bd-4adf-91a7-69aab161d09f, 35, Finished, Available, Finished)

<Spark SQL result set with 0 rows and 0 fields>

In [34]:
start_date='2015-01-01'
end_date='2030-12-31'

date_diff=spark.sql("select date_diff('{}','{}')".format(end_date,start_date)).first()[0]

StatementMeta(, fab74371-09bd-4adf-91a7-69aab161d09f, 36, Finished, Available, Finished)

In [35]:
date_diff

StatementMeta(, fab74371-09bd-4adf-91a7-69aab161d09f, 37, Finished, Available, Finished)

5843

now we will generate a range of numbers from 0 to 5843, use this to generate dates (adding these numbers to start date)

In [36]:
ID_Data=spark.range(0,date_diff)

StatementMeta(, fab74371-09bd-4adf-91a7-69aab161d09f, 38, Finished, Available, Finished)

In [37]:
ID_Data.show()

StatementMeta(, fab74371-09bd-4adf-91a7-69aab161d09f, 39, Finished, Available, Finished)

+---+
| id|
+---+
|  0|
|  1|
|  2|
|  3|
|  4|
|  5|
|  6|
|  7|
|  8|
|  9|
| 10|
| 11|
| 12|
| 13|
| 14|
| 15|
| 16|
| 17|
| 18|
| 19|
+---+
only showing top 20 rows



In [38]:
date_data = ID_Data.selectExpr("date_add('{}',cast(id as int)) as BusinessDate".format(start_date))

StatementMeta(, fab74371-09bd-4adf-91a7-69aab161d09f, 40, Finished, Available, Finished)

In [39]:
date_data.show()

StatementMeta(, fab74371-09bd-4adf-91a7-69aab161d09f, 41, Finished, Available, Finished)

+------------+
|BusinessDate|
+------------+
|  2015-01-01|
|  2015-01-02|
|  2015-01-03|
|  2015-01-04|
|  2015-01-05|
|  2015-01-06|
|  2015-01-07|
|  2015-01-08|
|  2015-01-09|
|  2015-01-10|
|  2015-01-11|
|  2015-01-12|
|  2015-01-13|
|  2015-01-14|
|  2015-01-15|
|  2015-01-16|
|  2015-01-17|
|  2015-01-18|
|  2015-01-19|
|  2015-01-20|
+------------+
only showing top 20 rows



In [40]:
date_data.createOrReplaceTempView("ViewDate")

StatementMeta(, fab74371-09bd-4adf-91a7-69aab161d09f, 42, Finished, Available, Finished)

In [41]:
%%sql
delete from Gold_Date;

StatementMeta(, fab74371-09bd-4adf-91a7-69aab161d09f, 43, Finished, Available, Finished)

<Spark SQL result set with 1 rows and 1 fields>

In [42]:
%%sql
insert into Gold_Date
select *, 
Year(BusinessDate) as Business_Year,
Month(BusinessDate) as Business_Month ,
Quarter(BusinessDate) as Business_Quarter ,
weekday(BusinessDate) as Business_Week  
from ViewDate

StatementMeta(, fab74371-09bd-4adf-91a7-69aab161d09f, 44, Finished, Available, Finished)

<Spark SQL result set with 0 rows and 0 fields>

In [43]:
%%sql
select * from Gold_Date

StatementMeta(, fab74371-09bd-4adf-91a7-69aab161d09f, 45, Finished, Available, Finished)

<Spark SQL result set with 1000 rows and 5 fields>